app/src/main/java/io/novafoundation/nova/app/root/presentation/deepLinks/ReferendumDeepLinkHandler.kt

```diff
@@ -0,0 +1,81 @@
+package io.novafoundation.nova.app.root.presentation.deepLinks
+
+import android.net.Uri
+import io.novafoundation.nova.feature_account_api.domain.interfaces.AccountRepository
+import io.novafoundation.nova.feature_governance_api.data.GovernanceStateUpdater
+import io.novafoundation.nova.feature_governance_impl.presentation.GovernanceRouter
+import io.novafoundation.nova.feature_governance_impl.presentation.referenda.details.ReferendumDetailsPayload
+import io.novafoundation.nova.runtime.ext.utilityAsset
+import io.novafoundation.nova.runtime.multiNetwork.ChainRegistry
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
+import io.novafoundation.nova.runtime.multiNetwork.chainsById
+import java.math.BigInteger
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.emptyFlow
+
+private const val GOV_DEEP_LINK_PREFIX = "/open/gov"
+
+class ReferendumDeepLinkHandler(
+    private val governanceRouter: GovernanceRouter,
+    private val chainRegistry: ChainRegistry,
+    private val governanceStateUpdater: GovernanceStateUpdater,
+    private val accountRepository: AccountRepository
+) : DeepLinkHandler {
+
+    override val callbackFlow: Flow<CallbackEvent> = emptyFlow()
+
+    override suspend fun matches(data: Uri): Boolean {
+        val userAccounts = accountRepository.allMetaAccounts()
+        val path = data.path ?: return false
+        val chainId = data.getChainId() ?: return false
+        val referendumId = data.getReferendumId()
+        val chainsById = chainRegistry.chainsById()
+        val chain = chainsById[chainId]
+        val governanceType = data.getGovernanceType()
```

Comment:
Looks like it pretty much dublicates the `handleDeeolink` logic. I suggest to only check that `path.startsWith(GOV_DEEP_LINK_PREFIX) == true` and remove the rest since it is handled by `handleDeepLink` anyways